### PR TITLE
EMTF DQM for emulator and data comparison

### PR DESCRIPTION
**Pull request by EsmaeelEskandari**

### CODE REVIEW

File: L1Trigger/L1TMuonEndCap/plugins/L1TMuonEndCapTrackProducer.cc [hunk 1]
```diff
@@ -65,13 +65,12 @@ void L1TMuonEndCapTrackProducer::produce(edm::Event& ev,
...
-  auto OutHits = std::make_unique<l1t::EMTFHitCollection>();
-  auto OutputTracks = std::make_unique<l1t::EMTFTrackExtraCollection>();
-  auto OutputHits = std::make_unique<l1t::EMTFHitExtraCollection>();
-  auto OutputHitsRPC = std::make_unique<l1t::EMTFHitExtraCollection>();
+  std::auto_ptr<l1t::RegionalMuonCandBxCollection > OutputCands (new l1t::RegionalMuonCandBxCollection);
+  std::auto_ptr<l1t::EMTFTrackCollection> OutTracks (new l1t::EMTFTrackCollection);
+  std::auto_ptr<l1t::EMTFHitCollection> OutHits (new l1t::EMTFHitCollection);
+  std::auto_ptr<l1t::EMTFTrackExtraCollection> OutputTracks (new l1t::EMTFTrackExtraCollection);
+  std::auto_ptr<l1t::EMTFHitExtraCollection> OutputHits (new l1t::EMTFHitExtraCollection);
+  std::auto_ptr<l1t::EMTFHitExtraCollection> OutputHitsRPC (new l1t::EMTFHitExtraCollection);
```
Comment (thomreis) [inline L1Trigger/L1TMuonEndCap/plugins/L1TMuonEndCapTrackProducer.cc]: @EsmaeelEskandari this PR changes the EMTF emulator which should not happen in a DQM related PR. If you want to change something in the emulator you should do a PR to cms-l1t-offline first, and @rekovic or @mulhearn will handle the PR to CMSSW.
In this case it looks like you are trying to undo a commit of @rekovic https://github.com/cms-sw/cmssw/commit/04ea88bdc6efbdea8c1f1a49f18209b9c2ba90b9 from 2 months ago.
Please check more carefully what changes are in your branch before making a PR.

Comment (thomreis) [inline L1Trigger/L1TMuonEndCap/plugins/L1TMuonEndCapTrackProducer.cc]: My guess is that you made this PR to 8_1_X from your 8_0_X development without taking into account the differences. The commit of Vladimir exists in 8_1_X but not in 8_0_X.

**Comment by EsmaeelEskandari** [comment]
​Thank you @thomreis.
I will undo the applied changes by git cherry-pick @rekovic 04ea88b.

**Comment by dmitrijus** [comment]
This needs a 90x port now. Please link me to it, in fact, most likely you can use the same branch as for 81x. Just create a pull request from EsmaeelEskandari:EMTF_DQM_emulator_data_comparison to CMSSW_9_0_X

**Comment by EsmaeelEskandari** [comment]
Dear @dmitrijus , I wanted to make the PR for 90x but it says that it will also contain some other 33 commits from other parts. Should I make this PR or I should use the last 90x IB in lxplus and git cms-merge-topic 16738 and push them to a new branch?

**Comment by dmitrijus** [comment]
Ah, I see. Looks like someone already diverged both branches... :(
What I would do, is:
```
1. git merge-base remotes/official-cmssw/CMSSW_8_1_X remotes/official-cmssw/CMSSW_9_0_X
2. git checkout <the id from above> -b EMTF_DQM_emulator_data_comparison_try2
3. do the cherry pick and force push the changes to this branch (of this PR)
5. create a PR for 90x, using the _same_ branch for both 81x and 90x.
```

If it fails or you get conflicts, or you don't understand what merge-base does, do it the old way "hard" way. Pick any 90x IB, apply changes and make a PR.

**Pull request closed without merging**